File: From_Last_Year/Jutge/Week 5/Sliding_window_2.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

/*
let L be an empty vector

SLIDING_WINDOW(S, frame, inc)
	return SLIDING_WINDOW_REC(S, frame, inc, 0)

SLIDING_WINDOW_REC(S, frame, inc, i)
	append seq[i:i+frame] to L
	
	while i <= |S|-frame then
		return SLIDING_WINDOW_REC(S, frame, inc, i+inc)
	
	return l

*/


vector <string> A;

vector <string> sw(string seq, int frame, int inc, int i);

vector <string> sw(string seq, int frame, int inc){
    return sw(seq, frame, inc, 0);
      
}

vector <string> sw(string seq, int frame, int inc, int i){
    
    if(i <= seq.length()-frame){
        A.push_back(seq.substr(i, frame));
        sw(seq, frame, inc, i+inc);
    }
    return A;
}    

int main()
{
    string seq;
    int frame;
    int inc;

    cin >> seq;
    cin >> frame;
    cin >> inc;

   for(auto &x : sw(seq, frame, inc)){
       cout << x << endl;
   }
   
    
}

File: From_Last_Year/Jutge/Week 5/Sliding_window_2.py
"""
let L be an empty list

SLIDING_WINDOW(S, frame, inc)
	return SLIDING_WINDOW_REC(S, frame, inc, 0)

SLIDING_WINDOW_REC(S, frame, inc, i)
	append seq[i:i+frame] to L
	
	while i <= |S|-frame-1 then
		return SLIDING_WINDOW_REC(S, frame, inc, i+inc)
	
	return l

"""
from jutge import read

seq = read(str)
frame = read(int)
inc = read(int)

l = []

def sliding_window(seq, frame, inc):
	
	return sliding_window_rec(seq, frame , inc, 0)


def sliding_window_rec(seq, frame, inc, i):
	l.append(seq[i:i+frame])

	while i <= len(seq)-frame-inc: 
	
		return sliding_window_rec(seq, frame, inc, i+inc)
	
	return l

for x in sliding_window(seq, frame, inc):
	print(x)
